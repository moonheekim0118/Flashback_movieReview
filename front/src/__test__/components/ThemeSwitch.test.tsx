import 'jsdom-global/register'; 
import React from "react";
import { mount } from 'enzyme';
import ThemeSwitch from '../../components/ThemeSwtich';

describe('<Badge/>', () => {
   let container = null;
   const mockFn = jest.fn();

   it('renders correctly', ()=>{ // 렌더링 테스트 
      container=mount(<ThemeSwitch onCheck={mockFn} checked={true}/>);
   });

   it('matches snapshot', ()=>{ // 스냅샷 비교 
    expect(container.html()).toMatchSnapshot();
   });

   it('should call onCheck if Switch is change',()=>{ // onClick 버튼 시뮬레이션 
    container.find('#switch').hostNodes().simulate('change');
    expect(mockFn).toHaveBeenCalled();
    });

})
