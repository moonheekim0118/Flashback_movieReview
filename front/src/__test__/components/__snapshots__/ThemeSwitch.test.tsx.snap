// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge/> matches snapshot 1`] = `"<div class=\\"ThemeSwtich__Container-sc-1140ti1-0 daFpoA\\"><div style=\\"position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 12px; transition: opacity 0.25s; user-select: none;\\"><div class=\\"react-switch-bg\\" style=\\"height: 24px; width: 48px; margin: 0px; position: relative; background: rgb(153, 153, 153); border-radius: 12px; cursor: pointer; transition: background 0.25s;\\"><div style=\\"height: 24px; width: 26px; position: relative; opacity: 1; pointer-events: none; transition: opacity 0.25s;\\"><svg height=\\"100%\\" width=\\"100%\\" viewBox=\\"-2 -5 17 21\\" style=\\"position: absolute; top: 0px;\\"><path d=\\"M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0\\" fill=\\"#fff\\" fill-rule=\\"evenodd\\"></path></svg></div><div style=\\"height: 24px; width: 26px; position: absolute; opacity: 0; right: 0px; top: 0px; pointer-events: none; transition: opacity 0.25s;\\"><svg viewBox=\\"-2 -5 14 20\\" height=\\"100%\\" width=\\"100%\\" style=\\"position: absolute; top: 0px;\\"><path d=\\"M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12\\" fill=\\"#fff\\" fill-rule=\\"evenodd\\"></path></svg></div></div><div class=\\"react-switch-handle\\" style=\\"height: 22px; width: 22px; background: rgb(40, 44, 53); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(25px); top: 1px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;\\"></div><input type=\\"checkbox\\" role=\\"switch\\" style=\\"border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;\\" id=\\"switch\\" checked=\\"\\"></div></div>"`;
